prive/generators/generator.py

```diff
@@ -33,15 +39,55 @@ def fit(self, dataset):
         self.dataset = dataset
         self.trained = True
 
-    def generate(self, num_samples):
-        return self.dataset.sample(num_samples)
-
-    def __call__(self, dataset, num_samples):
+    def generate(self, num_samples, random_state = None):
+        if self.trained: 
+            return self.dataset.sample(num_samples, random_state = random_state)
+        else:
+            raise RuntimeError("No dataset provided to generator")
+        
+    def __call__(self, dataset, num_samples, random_state = None):
         self.fit(dataset)
-        return self.generate(num_samples)
+        return self.generate(num_samples, randon_state = random_state)
 
 
 # And importantly, import generators from disk executables.
 
 class GeneratorFromExecutable(Generator):
-	"""This class interfaces with a generator as executable on disk."""
\ No newline at end of file
+    """
+    A class which wraps an external executable as a generator. Currently supports
+    only tabular datasets.
+    """
+    def __init__(self, exe):
+        """
+        Parameters
+        ----------
+        exe : The path to the executable as a string.
+        """
+        actual_exe = shutil.which(exe)
+        if actual_exe is not None:
+            self.exe = actual_exe
+        else: 
+            actual_exe = shutil.which(exe, path = os.getcwd())
+            if actual_exe is not None:
+                self.exe = actual_exe
+            else:
+                raise RuntimeError("Can't find user-supplied executable")
+        super().__init__()
+
+    def fit(self, dataset):
+        self.dataset = dataset
+        self.trained = True
+
+    def generate(self, num_samples):
```

Comment:
It seems that this is the right place to ensure fixed randomness, by adding a `random_state` argument here. But it's not clear to me how to make this interface with the existing exe we have, `raw`.